lib/src/test/kotlin/org/konigsoftware/kontext/KonigKontextClientInterceptorTest.kt

```diff
@@ -0,0 +1,257 @@
+
+class KonigKontextClientInterceptorTest {
+    @Rule
+    val grpcCleanup = GrpcCleanupRule()
+
+    private object TestProtobufKontextKey : KonigKontextProtobufKey<HelloRequest>(HelloRequest::class)
+
+    private object TestCustomKontextKey : KonigKontextKey<String>() {
+        override val defaultValue = ""
+
+        override fun valueFromBinary(binaryValue: ByteArray): String = String(binaryValue)
+
+        override fun valueToBinary(value: String): ByteArray = value.toByteArray()
+    }
+
+    // Test interceptor used to assert the headers captured on an incoming request
+    private class TestServerInterceptor : ServerInterceptor {
+        var capturedMetadata: Metadata? = null
+
+        override fun <ReqT : Any?, RespT : Any?> interceptCall(
+            call: ServerCall<ReqT, RespT>?,
+            headers: Metadata,
+            next: ServerCallHandler<ReqT, RespT>
+        ): Listener<ReqT> {
+            capturedMetadata = headers
+
+            return next.startCall(call, headers)
+        }
+    }
+
+    @Test
+    fun `Given downstream rpc, when calling rpc from KonigKontext intercepted client and the protobuf KonigKontext is empty, then the KonigKontextKey default value is included on request headers`() {
+        val testServerInterceptor = TestServerInterceptor()
+
+        // Create a fake in process server
+        val serverName = InProcessServerBuilder.generateName()
+        grpcCleanup.register(
+            InProcessServerBuilder.forName(serverName).directExecutor()
+                .addService(ServerInterceptors.intercept(object : GreeterImplBase() {
+                    override fun sayHello(request: HelloRequest?, responseObserver: StreamObserver<HelloReply>?) {
+                        val response = HelloReply.getDefaultInstance()
+                        responseObserver?.onNext(response)
+                        responseObserver?.onCompleted()
+                    }
+                }, testServerInterceptor)).build()
+                .start()
+        )
+
+        // Create a client channel and register for automatic graceful shutdown.
+        val channel = grpcCleanup.register(InProcessChannelBuilder.forName(serverName).directExecutor().build())
+        // Register KonigKontextClientInterceptor
+        val blockingStub =
+            GreeterGrpc.newBlockingStub(channel).withKonigKontextInterceptor(TestProtobufKontextKey)
+
+        blockingStub.sayHello(HelloRequest.getDefaultInstance())
+
+        assertNotNull(testServerInterceptor.capturedMetadata)
+        assertNotNull(testServerInterceptor.capturedMetadata!!.get(TestProtobufKontextKey.grpcHeaderKey))
+        assert(HelloRequest.getDefaultInstance().toByteArray().contentEquals(testServerInterceptor.capturedMetadata!!.get(TestProtobufKontextKey.grpcHeaderKey)))
+        assert(TestProtobufKontextKey.valueToBinary(HelloRequest.getDefaultInstance()).contentEquals(testServerInterceptor.capturedMetadata!!.get(TestProtobufKontextKey.grpcHeaderKey)))
```

Comment:
Not sure I am following how this relates to the line above

(same in other test)

Reply:
I guess its just a sanity check that the value set on the grpc header equals both `HelloRequest().getDefaultInstance().toByteArray()` and `TestProtobufKontextKey.valueToBinary(HelloRequest.getDefaultInstance())`. So I guess its a bit redundant